lib/nerves_runtime/heart.ex

```diff
@@ -92,28 +118,62 @@ defmodule Nerves.Runtime.Heart do
     _ -> :error
   end
 
+  # v1 and v2 parsers
   defp parse_attribute(["program_name", str]), do: {:program_name, str}
   defp parse_attribute(["program_version", str]), do: {:program_version, Version.parse!(str)}
+
+  defp parse_attribute(["heartbeat_timeout", str]),
+    do: {:heartbeat_timeout, atoi(str)}
+
+  # v1 parsers
   defp parse_attribute(["identity", str]), do: {:identity, str}
-  defp parse_attribute(["firmware_version", str]), do: {:firmware_version, String.to_integer(str)}
-  defp parse_attribute(["options", "0x" <> hex]), do: {:options, String.to_integer(hex, 16)}
+  defp parse_attribute(["firmware_version", str]), do: {:firmware_version, atoi(str)}
+  defp parse_attribute(["options", "0x" <> hex]), do: {:options, atoi(hex, 16)}
   defp parse_attribute(["options", option_list]), do: {:options, parse_option_list(option_list)}
-  defp parse_attribute(["time_left", str]), do: {:time_left, String.to_integer(str)}
-  defp parse_attribute(["pre_timeout", str]), do: {:pre_timeout, String.to_integer(str)}
-  defp parse_attribute(["timeout", str]), do: {:timeout, String.to_integer(str)}
+  defp parse_attribute(["time_left", str]), do: {:time_left, atoi(str)}
+  defp parse_attribute(["pre_timeout", str]), do: {:pre_timeout, atoi(str)}
+  defp parse_attribute(["timeout", str]), do: {:timeout, atoi(str)}
   defp parse_attribute(["last_boot", str]), do: {:last_boot, parse_last_boot(str)}
 
-  defp parse_attribute(["heartbeat_timeout", str]),
-    do: {:heartbeat_timeout, String.to_integer(str)}
+  # v2 parsers
+  defp parse_attribute(["wdt_identity", str]), do: {:wdt_identity, str}
+  defp parse_attribute(["wdt_firmware_version", str]), do: {:wdt_firmware_version, atoi(str)}
+  defp parse_attribute(["wdt_options", "0x" <> hex]), do: {:wdt_options, atoi(hex, 16)}
+
+  defp parse_attribute(["wdt_options", option_list]),
+    do: {:wdt_options, parse_option_list(option_list)}
+
+  defp parse_attribute(["wdt_pet_time_left", str]), do: {:wdt_pet_time_left, atoi(str)}
+  defp parse_attribute(["wdt_pre_timeout", str]), do: {:wdt_pre_timeout, atoi(str)}
+  defp parse_attribute(["wdt_timeout", str]), do: {:wdt_timeout, atoi(str)}
+  defp parse_attribute(["wdt_time_left", str]), do: {:wdt_time_left, atoi(str)}
+  defp parse_attribute(["wdt_last_boot", str]), do: {:wdt_last_boot, parse_last_boot(str)}
+  defp parse_attribute(["heartbeat_time_left", str]), do: {:heartbeat_time_left, atoi(str)}
+  defp parse_attribute(["init_handshake_timeout", str]), do: {:init_handshake_timeout, atoi(str)}
 
+  defp parse_attribute(["init_handshake_time_left", str]),
+    do: {:init_handshake_time_left, atoi(str)}
+
+  defp parse_attribute(["init_handshake_happened", str]),
+    do: {:init_handshake_happened, parse_bool(str)}
+
+  # unknowns
   defp parse_attribute([_unknown, _str]), do: nil
   defp parse_attribute([""]), do: nil
 
+  # helpers
+  defp atoi(str), do: String.to_integer(str)
+  defp atoi(str, base), do: String.to_integer(str, base)
```

Comment:
I love the C really shinning through here 🤣

Reply:
Yeah... I did it to keep a few lines from wrapping.